### PR TITLE
Kernel+tail: Handle truncation in follow mode

### PR DESCRIPTION
**Pull request by fdellwing**
This is work towards #23097

CC: @ADKaster

### CODE REVIEW

File: Kernel/FileSystem/InodeFile.cpp [hunk 1]
```diff
@@ -96,6 +96,8 @@ ErrorOr<NonnullOwnPtr<KString>> InodeFile::pseudo_path(OpenFileDescription const
 ErrorOr<void> InodeFile::truncate(u64 size)
 {
     TRY(m_inode->truncate(size));
+    // FIXME: In most cases, we already updated the timestamps in Inode::truncate.
+    //        However, we should probably do it here as well, to ensure that the timestamps are always updated.
```
Comment (gmta) [inline Kernel/FileSystem/InodeFile.cpp:100]: What's the FIXME exactly? We already update the timestamps two lines down.

File: Userland/Utilities/tail.cpp [hunk 2]
```diff
@@ -102,12 +103,32 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
 
         Core::EventLoop event_loop;
         auto watcher = TRY(Core::FileWatcher::create());
         watcher->on_change = [&](Core::FileWatcherEvent const& event) {
             if (event.type == Core::FileWatcherEvent::Type::ContentModified) {
+                auto maybe_current_size = f->size();
+                if (maybe_current_size.is_error()) {
+                    auto error = maybe_current_size.release_error();
+                    warnln(error.string_literal());
+                    event_loop.quit(error.code());
```
Comment (gmta) [inline Userland/Utilities/tail.cpp:117]: This error handling code is repeated 4 times, maybe put it in a lambda?

Comment (fdellwing) [reply Userland/Utilities/tail.cpp:117]: lambdas are not my strong suite, but I'll take a look.

File: Kernel/FileSystem/InodeFile.cpp (hunk 1 above)
Comment (fdellwing) [reply Kernel/FileSystem/InodeFile.cpp:100]: So, should I change the `FIXME` into a `NOTE`? Or write something like "Make sure that the timestamps are updated by Inode::truncate for all filesystems before removing it here"?

Comment (gmta) [reply Kernel/FileSystem/InodeFile.cpp:100]: That last one sounds fine!

**Review by ADKaster — APPROVED** [review]

**Pull request merged**